### PR TITLE
Increase iteration count because of reports of local failures 

### PR DESCRIPTION
**Pull request by apaleyes**
MES integ test was reported to fail on a number of local machines (@sebastianober , @uri-granta ) while mysteriously passing GitHub Actions build. This commit shall unblock local development, while we shall also create an issue to investigate this further. Note: on my machine it converges after 22 iterations, I've also heard about 21 iterations elsewhere, so the count is increased a bit excessively to allow for local fluctuations.

Also added local .vscode folder to .gitignore.

### CODE REVIEW

File: tests/integration/test_bayesian_optimization.py [hunk 1]
```diff
@@ -68,7 +68,7 @@
             (20, EfficientGlobalOptimization()),
             (25, EfficientGlobalOptimization(AugmentedExpectedImprovement().using(OBJECTIVE))),
             (
-                15,
+                25,
```
Comment (uri-granta) [inline tests/integration/test_bayesian_optimization.py]: I think it also works with this:

```suggestion
                21,
```

**Review by uri-granta — APPROVED** [review]
I took a quick look, and it looks like this is a random number generator issue revealed by the GIBBON refactoring (https://github.com/secondmind-labs/trieste/pull/349) which changed the order of some initialisation. Not sure why the RNG is behaving differently on GH, but since the integration tests should ideally pass regardless of the RNG seed I think this fix is probably best.

File: tests/integration/test_bayesian_optimization.py (hunk 1 above)
Comment (apaleyes) [reply tests/integration/test_bayesian_optimization.py]: It only works with 22 on my machine.

Comment (apaleyes) [reply tests/integration/test_bayesian_optimization.py]: updated to 22 as the min known to fix all local failures

**Pull request merged**